drivers/net/ethernet/renesas/rswitch.c

```diff
@@ -3506,12 +3563,27 @@ static int rswitch_bpool_config(struct rswitch_private *priv)
 
 static void rswitch_queue_interrupt(struct rswitch_device *rdev)
 {
-	if (napi_schedule_prep(&rdev->napi)) {
-		rswitch_enadis_data_irq(rdev->priv, rdev->tx_chain->index, false);
-		rswitch_enadis_data_irq(rdev->priv, rdev->rx_default_chain->index, false);
-		if (rdev->rx_learning_chain)
-			rswitch_enadis_data_irq(rdev->priv, rdev->rx_learning_chain->index, false);
-		__napi_schedule(&rdev->napi);
+	if (!rdev->mondev) {
+		if (napi_schedule_prep(&rdev->napi)) {
+			rswitch_enadis_data_irq(rdev->priv, rdev->tx_chain->index, false);
+			rswitch_enadis_data_irq(rdev->priv, rdev->rx_default_chain->index, false);
+			if (rdev->rx_learning_chain)
+				rswitch_enadis_data_irq(rdev->priv, rdev->rx_learning_chain->index, false);
+			__napi_schedule(&rdev->napi);
+		}
+	} else {
+		struct rswitch_private *priv = rdev->priv;
+
+		/* Schedule napi for all rmon devices as
+		 * they share the same chain.
+		 */
+		list_for_each_entry(rdev, &priv->rdev_list, list) {
```

Comment:
Unsafe list access

Reply:
This part of code was reworked and now there is no need to lock list as rmon devices are stored in a separate array.